src/ValueParsers/TimeParser.php

```diff
@@ -137,61 +168,22 @@ private function getPrecisionFromYear( $year ) {
 	}
 
 	/**
-	 * @param string $value
-	 *
-	 * @return array with the following keys.
-	 *            sign, year, month, day, hour, minute, second, calendar
-	 *
-	 * @throws InvalidArgumentException
-	 * @throws ParseException
-	 */
-	private function splitTimeString( $value ) {
-		if ( !is_string( $value ) ) {
-			throw new InvalidArgumentException( '$value must be a string' );
-		}
-
-		$pattern = '@^'
-			. '\s*' . '([\+\-]?)'
-			. '\s*' . '(\d{1,16})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})Z'
-			. '\s*\(?\s*' . CalendarModelParser::MODEL_PATTERN . '\s*\)?'
-			. '\s*$@iu';
-
-		if ( !preg_match( $pattern, $value, $groups ) ) {
-			throw new ParseException( 'Malformed time', $value, self::FORMAT_NAME );
-		}
-
-		return array(
-			'sign' => $groups[1],
-			'year' => $groups[2],
-			'month' => $groups[3],
-			'day' => $groups[4],
-			'hour' => $groups[5],
-			'minute' => $groups[6],
-			'second' => $groups[7],
-			'calendar' => $groups[8],
-		);
-	}
-
-	/**
-	 * @param array $timeParts with the following keys.
-	 *            sign, year, month, day, hour, minute, second, calendar
-	 *
-	 * @throws InvalidArgumentException
+	 * @param string $calendarModelName
 	 *
 	 * @return string
 	 */
-	private function getTimeStringFromParts( array $timeParts ) {
-		if( array_keys( $timeParts ) !== array( 'sign', 'year', 'month', 'day', 'hour', 'minute', 'second', 'calendar' ) ) {
-			throw new InvalidArgumentException( 'Time string can not be created with missing $timeParts keys' );
+	private function getCalendarModel( $calendarModelName ) {
+		if ( !empty( $calendarModelName ) ) {
+			return $this->calendarModelParser->parse( $calendarModelName );
 		}
-		return $timeParts['sign']
-			. $timeParts['year'] . '-'
-			. $timeParts['month'] . '-'
-			. $timeParts['day'] . 'T'
-			. $timeParts['hour'] . ':'
-			. $timeParts['minute'] . ':'
-			. $timeParts['second'] . 'Z';
 
+		// The calendar model is an URI and URIs can't be case-insensitive
+		switch ( $this->getOption( self::OPT_CALENDAR ) ) {
+			case self::CALENDAR_JULIAN:
+				return self::CALENDAR_JULIAN;
+			default:
+				return self::CALENDAR_GREGORIAN;
```

Comment:
So, if an unknown calendar was set, we silently replace it by Gregorian?! That seems wrong. In such a case, we should fail, because we don't know how to parse such a date.